jcl/src/java.base/share/classes/openj9/internal/criu/InternalCRIUSupport.java

```diff
@@ -35,7 +36,8 @@
 /*[ENDIF] JAVA_SPEC_VERSION >= 17 */
 public final class InternalCRIUSupport {
 	/*[IF CRIU_SUPPORT]*/
-	private static boolean criuSupportEnabled;
+	// three states: -1 for unknown, 1 for enabled, 0 for disabled
+	private static int criuSupportEnabled = -1;
 	private static boolean nativeLoaded;
 	private static boolean initComplete;
 
```

Comment:
both of these natives can be moved to the JVM, they dont require libcriu to be loaded

Reply:
we also need a public `isCheckpointAllowed` in CRIUSupport

Reply:
Moved `InternalCRIUSupport.isCheckpointAllowedImpl()` from `j9criu` to `jclse` as well.
Added `CRIUSupport. isCheckpointAllowed()`.